### PR TITLE
TheWire rework UI, and update Data Messages.

### PR DESCRIPTION
**Pull request by drbob**
Significant rework of The Wire to support:
 * A interface closer to Twitter.
 * Replies / Republishes / Likes
 * Up to 4 Images

### CODE REVIEW

**Comment by defnax** [comment]
i like this design lots better then the other one :)

File: libretroshare/src/services/p3wire.cc [hunk 1]
```diff
@@ -31,6 +31,23 @@
 RsWire *rsWire = NULL;
 
 
+uint32_t RsWirePulse::ImageCount()
+{
+	if (!mImage4.empty()) {
```
Comment (csoler) [inline libretroshare/src/services/p3wire.cc]: The logic seems risky here. One can craft a msg with mImage3=nullptr and mImage4 not empty. But still the number of images will not be 4.

**Comment by defnax** [comment]
at the moment i doesnt like the Show Group & Show Parent Buttons which not needed there
best we need a Back Button on the same place like on twitter on top left
image

@drbob i have a little pr changes for the group item
https://github.com/drbob/RetroShare/pull/1

image

**Comment by drbob** [comment]
Defnax: I like the idea of a back button + history, I'll try and add this in the next PR.
I think the "Show Parent" and "Show Group" are necessary - just to allow fluid navigation.
Obviously, I'll change the buttons into clickable Labels, in the future.
PS. I've merged your PR.

File: libretroshare/src/services/p3wire.cc (hunk 1 above)
Comment (drbob) [reply libretroshare/src/services/p3wire.cc]: made this more robust

**Pull request merged**